spec/lib/govuk_elements_form_builder/form_builder_spec.rb

```diff
@@ -9,55 +9,37 @@ class TestHelper < ActionView::Base; end
   let(:builder) { described_class.new resource.to_sym, resource_form, helper, {} }
   let(:parsed_html) { Nokogiri::HTML(subject) }
 
-  describe 'govuk_text_field' do
-    let(:attribute) { 'email' }
-    let(:params) { [:email] }
-    let(:email_label) { I18n.t("activemodel.attributes.#{resource}.#{attribute}") }
-    let(:email_hint) { I18n.t("helpers.hint.#{resource}.#{attribute}") }
-    let(:input) { parsed_html.at_css("input##{attribute}") }
-
-    subject { builder.govuk_text_field(*params) }
-
-    it 'generates a text field' do
-      expect(input.classes).to include('govuk-input')
-      expect(input[:type]).to eq('text')
-      expect(input[:name]).to eq("#{resource}[#{attribute}]")
-    end
-
+  shared_examples_for 'a basic input field' do
     it 'surrounds the field in a div' do
-      div = input.parent
-
+      div = tag.parent
       expect(div.name).to eq('div')
       expect(div.classes).to include('govuk-form-group')
     end
 
     it 'includes a label' do
-      label = input.parent.at_css('label')
-
       expect(label.classes).to include('govuk-label')
       expect(label[:for]).to eq('email')
```

Comment:
This shared example would be a lot more flexible if every instance of `'email'` was replaced with `attribute`. Note that `attribute` is already used in the hint sections of this shared example.